redis/src/main/scala/zio/redis/Redis.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2021 John A. De Goes and the ZIO contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.redis
+
+import zio.{Has, URLayer, ZIO, ZLayer}
+import zio.schema.codec.Codec
+
+object Redis {
+  trait Service {
+    def codec: Codec
+    def executor: RedisExecutor.Service
+  }
+
+  lazy val live: URLayer[RedisExecutor with Has[Codec], Redis] =
+    ZLayer.identity[RedisExecutor] ++ ZLayer.identity[Has[Codec]] >>> RedisService
+
+  private[this] final val RedisService =
+    ZLayer
+      .fromServicesManaged[RedisExecutor.Service, Codec, Any, Nothing, Redis.Service] { (e, c) =>
+        ZIO
+          .succeed(new Service {
+            override val codec: Codec                    = c
+            override val executor: RedisExecutor.Service = e
+          })
+          .toManaged_
+      }
```

Comment:
This can be simplified:

```suggestion
    ZIO.services[Codec, RedisExecutor.Service].map { env =>
      new Service {
        val codec: Codec = c
        val executor: RedisExecutor.Service = e
      }
    }.toLayer
```